src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,147 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_LENGTH = 1 << 4;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int actualLoadFactorNumber;
+    private int size;
+    private int newLength;
+    private Node<K, V>[] table;
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V>[] oldTab = table;
+        if (table == null || size >= actualLoadFactorNumber) {
+            oldTab = resize();
+        }
+        int hash = hash(key);
+        int index = Math.abs(hash(key) % newLength);
+        table = addByIndex(hash, key, value, oldTab, index);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (size == 0) {
+            return null;
+        }
+        int index = Math.abs(hash(key) % newLength);
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (key == null && node.key == null || node.key != null && node.key.equals(key)) {
+                return node.value;
+            }
+        }
+        return table[index].value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K,V> [] addByIndex(int hash, K key, V value, Node<K, V>[] oldTab, int index) {
+        Node<K, V> newNode = new Node<>(hash, key, value, null);
+        Node<K, V> currentBucket = oldTab[index];
+        if (currentBucket == null) {
+            oldTab[index] = newNode;
+            size++;
+        } else {
+            Node<K, V> prevNode = null;
+            while (currentBucket != null) {
+                if (Objects.equals(currentBucket.key, key)) {
+                    currentBucket.value = value;
+                    return oldTab;
+                }
+                prevNode = currentBucket;
+                currentBucket = currentBucket.next;
+            }
+            prevNode.next = newNode;
+            size++;
+        }
+        return oldTab;
+    }
+
+    private int hash(Object key) {
+        int h;
+        return (key == null) ? 0 : (h = key.hashCode()) ^ (h >>> 16);
+    }
```

Comment:
make this method to be getIndex(K key)
you use this logic only to have proper index so let's do it here